documentation/api/v1/host_api_doc.rb

```diff
@@ -0,0 +1,236 @@
+require 'swagger/blocks'
+
+module HostApiDoc
+  include Swagger::Blocks
+
+  HOST_DESC = 'The IP address of the host.'
+  HOST_EXAMPLE = '127.0.0.1'
+  MAC_DESC = 'MAC Address of the host'
+  MAC_EXAMPLE = 'AA:BB:CC:11:22:33'
+  COMM_DESC = 'Unused attribute.'
+  NAME_DESC = 'Hostname of the host.'
+  NAME_EXAMPLE = 'domain_controller'
+  STATE_DESC = 'The last seen connectivity state of this host.'
+  OS_NAME_DESC = 'The name of the operating system.'
+  OS_NAME_EXAMPLE = "'Windows XP', 'Ubuntu', or 'Mac OS X'"
+  OS_FLAVOR_DESC = 'The flavor of operating system.'
+  OS_FLAVOR_EXAMPLE = "'Enterprise', 'Pro', or 'Home'"
+  OS_SP_DESC = 'The service pack version the operating system is running.'
+  OS_SP_EXAMPLE = "'SP2'"
+  OS_LANG_DESC = 'The language the operating system is using.'
+  OS_LANG_EXAMPLE = "'English', 'French', or 'en-US'"
+  OS_FAMILY_DESC = 'The major family the operating system belongs to.'
+  OS_FAMILY_EXAMPLE = "'Windows', 'Linux', or 'OS X'"
+  ARCH_DESC = 'The architecture of the host\'s CPU OR the programming language for virtual machine programming language like Ruby, PHP, and Java.'
+  DETECTED_ARCH_DESC = 'The architecture of the host\'s CPU as detected by `Recog`. If arch is not \'unknown\', this is undefined.'
+  PURPOSE_DESC = 'The main function of the host.'
+  INFO_DESC = 'Customizable information about the host.'
+  COMMENTS_DESC = 'A place for storing notes or findings about the host.'
+  SCOPE_DESC = 'Interface identifier for link-local IPv6.'
+  VIRTUAL_HOST_DESC = 'The name of the virtualization software.'
+  VIRTUAL_HOST_EXAMPLE = "'VMWare', 'QEMU', 'Xen', or 'Docker'"
+  NOTE_COUNT_DESC = 'Cached count of the number of associated notes.'
+  VULN_COUNT_DESC = 'Cached count of the number of associated vulns.'
+  SERVICE_COUNT_DESC = 'Cached count of the number of associated services.'
+  HOST_DETAIL_COUNT_DESC = 'Cached count of the number of associated host details.'
+  EXPLOIT_ATTEMPT_COUNT_DESC = 'Cached count of the number of associated exploit attempts.'
+  CRED_COUNT_DESC = 'Cached count of the number of associated creds.'
+  STATE_ENUM = [ 'alive', 'down', 'unknown' ]
+  ARCH_ENUM = [
```

Comment:
Where did this arch list come from? It looks a bit different from the one in [models/mdm/host.rb](https://github.com/rapid7/metasploit_data_models/blob/master/app/models/mdm/host.rb).

Reply:
Should we be providing this full list for host given the host model will actually set `:arch` to Unknown and `:detected_arch` to the passed in value if it is not in the array declared in `models/mdm/host.rb`? (See `normalize_arch` method)

Reply:
Where is `normalize_arch` located at? The only one I can find is in in `modules/exploits/android/browser/webview_addjavascriptinterface.rb` and it doesn't seem to behave the way you described?

Reply:
The method is in [metasploit_data_models/app/models/mdm/host.rb](https://github.com/rapid7/metasploit_data_models/blob/ac7d3214ec02c915c8eee58412e0e92a02db54e2/app/models/mdm/host.rb#L609)